src/openapi/response.rs

```diff
@@ -114,14 +114,178 @@ impl ResponseBuilder {
     }
 }
 
+/// Trait with convenience functions for documenting response bodies.
+///
+/// This trait requires a feature-flag to enable:
+/// ```text
+/// [dependencies]
+/// utoipa = { version = "1", features = ["openapi_extensions"] }
+/// ```
+///
+/// Once enabled, with a single method call we can add [`Content`] to our ResponseBuilder
+/// that references a responses (or component) schema using content-tpe "application/json":
+///
+/// ```rust
+/// use utoipa::openapi::response::{ResponseBuilder, ResponseExt};
+///
+/// let request = ResponseBuilder::new()
+///     .description("A sample response")
+///     .json_response_ref("MyResponsePayload").build();
+/// // Alternately for component, use
+/// // let request = ResponseBuilder::new().json_component_ref("MyResponsePayload").build();
+/// ```
+///
+/// If serialized to JSON, the above will result in a response schema like this:
+///
+/// ```json
+/// {
+///   "description": "A sample response",
+///   "content": {
+///     "application/json": {
+///       "schema": {
+///         "$ref": "#/components/responses/MyResponsePayload"
+///       }
+///     }
+///   }
+/// }
+/// ```
+///
+#[cfg(feature = "openapi_extensions")]
+pub trait ResponseExt {
+    fn json_component_ref(self, ref_name: &str) -> Self;
+    fn json_response_ref(self, ref_name: &str) -> Self;
+}
+
+
+#[cfg(feature = "openapi_extensions")]
+impl ResponseExt for Response {
+    /// Add [`Content`] to [`Response`] referring to a component-schema
+    /// with Content-Type `application/json`.
```

Comment:
Not sure if this doc comment will be shown in docs actually when it is placed over the implementation.  Same for the other instances below :point_down:  If you run this command `cargo doc --workspace --no-deps --features actix_extras,json,debug` and see from the docs whether it is there. I think this doc comment needs to be placed on the trait function itself [line 155](https://github.com/juhaku/utoipa/pull/240/files#diff-eea3ccce264a9afded89ef39496101cb1a5bf8c1fedc4cd4ddf9221e4744d922R155).